### ISSUE TITLE
sql: add full support for window frames

### ISSUE THREAD

**Issue by yuzefovich** [cockroachdb/cockroach#27100]
PG release 11 plans to "support all options shown in the SQL:2011 standard", so we should do the same.

Main additions are support of `GROUPS` mode of framing, optional `frame_exclusion` clause, and optional `FILTER`. See https://www.postgresql.org/docs/11/static/sql-expressions.html#SYNTAX-WINDOW-FUNCTIONS for reference.

**Comment by yuzefovich** [comment]
I will close this issue once #28262 is merged since all things I planned to implement with this issue will have been done. For a request for additional support of the specification, please open a new issue.
